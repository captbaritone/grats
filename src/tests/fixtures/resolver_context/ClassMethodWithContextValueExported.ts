/** @gqlContext */
export type GratsContext = {
  greeting: string;
};

/** @gqlType */
export class SomeType {
  /** @gqlField */
  greeting(ctx: GratsContext): string {
    return ctx.greeting;
  }
}
