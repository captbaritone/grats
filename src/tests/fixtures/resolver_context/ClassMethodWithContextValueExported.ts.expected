-----------------
INPUT
----------------- 
/** @gqlContext */
export type GratsContext = {
  greeting: string;
};

/** @gqlType */
export class SomeType {
  /** @gqlField */
  greeting(ctx: GratsContext): string {
    return ctx.greeting;
  }
}

-----------------
OUTPUT
-----------------
-- SDL --
type SomeType {
  greeting: String @metadata
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const SomeTypeType: GraphQLObjectType = new GraphQLObjectType({
        name: "SomeType",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString,
                    resolve(source, _args, context) {
                        return source.greeting(context);
                    }
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [SomeTypeType]
    });
}
